tools/payments/client.go

```diff
@@ -0,0 +1,185 @@
+		if err != nil {
+			return nil, fmt.Errorf("failed to read test-mode ca.crt: %w", err)
+		}
+		certPool.AppendCertsFromPEM(pem)
+		tlsConfig.RootCAs = certPool
+	}
+
+	rc := &redisClient{
+		env: env,
+		redis: redis.NewClusterClient(
+			&redis.ClusterOptions{
+				Addrs: addrs, Password: pass, Username: username,
+				DialTimeout:     15 * time.Second,
+				WriteTimeout:    5 * time.Second,
+				MaxRetries:      5,
+				MinRetryBackoff: 5 * time.Millisecond,
+				MaxRetryBackoff: 500 * time.Millisecond,
+				PoolSize:        10,
+				PoolTimeout:     30 * time.Second,
+				TLSConfig:       tlsConfig,
+			}),
+	}
+	err := rc.redis.Ping(ctx).Err()
+	if err != nil {
+		return nil, fmt.Errorf("failed to ping redis: %w", err)
+	}
+	return rc, nil
+}
+
+// ConfigureWorker implements settlement client
+func (rc *redisClient) ConfigureWorker(ctx context.Context, stream string, config *WorkerConfig) error {
+	_, err := rc.redis.XAdd(
+		ctx, &redis.XAddArgs{
+			Stream: stream,
+			Values: map[string]interface{}{
+				"data": config}},
+	).Result()
+	if err != nil {
+		return fmt.Errorf("failed to push config to workers: %w", err)
+	}
+	return nil
+}
+
+// PrepareTransaction implements settlement client
+func (rc *redisClient) PrepareTransaction(ctx context.Context, t *PrepareTx) error {
+	body, err := json.Marshal(t)
+	if err != nil {
+		return fmt.Errorf("failed to serialize transaction: %w", err)
+	}
+
+	// message wrapper for prepare
+	message := &prepareWrapper{
+		ID:        uuid.New(),
+		Timestamp: time.Now(),
+		Body:      string(body),
+	}
+
+	// add to stream
+	_, err = rc.redis.XAdd(
+		ctx, &redis.XAddArgs{
+			Stream: prepareStream,
+			Values: map[string]interface{}{
+				"data": message}},
+	).Result()
+	if err != nil {
+		return fmt.Errorf("failed to prepare transaction: %w", err)
+	}
+	return nil
+}
+
+// SubmitTransaction implements settlement client
+func (rc *redisClient) SubmitTransaction(ctx context.Context, signer httpsignature.ParameterizedSignator, at *AttestedTx) error {
+	body, err := json.Marshal(at)
+	if err != nil {
+		return fmt.Errorf("failed to marshal attested transaction body: %w", err)
+	}
+	// create a request
+	req, err := http.NewRequest("POST", rc.env+"/authorize", bytes.NewBuffer(body))
+	if err != nil {
+		return fmt.Errorf("failed to create request to sign: %w", err)
+	}
+	// we will be signing, need all these headers for it to go through
+	req.Header.Set("Host", rc.env)
+	req.Header.Set("Digest", fmt.Sprintf("%x", sha256.Sum256(body)))
+	req.Header.Set("Content-Length", fmt.Sprintf("%d", len(body)))
+	req.Header.Set("Content-Type", "application/json")
+
+	// http sign the request
+	err = signer.SignRequest(req)
+	if err != nil {
+		return fmt.Errorf("failed to sign request: %w", err)
+	}
+
+	// populate the submitWrapper for submission
+	message := &submitWrapper{
+		ID:            uuid.New(),
+		Timestamp:     time.Now(),
+		Host:          req.Header.Get("Host"),
```

Comment:
could we change the message format to match that in the [event.Message](https://github.com/brave-intl/bat-go/pull/1361/files#diff-333220128775a94b9b577fc04c2b8fc254914c3d260e40ade449fd2a46605e7bR27) this will be a lib and can be reused. This would become

```
        // Headers map[string]string
        
	headers := make(Headers)
	headers["Host"] = "some-host"
	headers["Digest"] = "some-digest"

	message := Message{
		ID:        uuid.NewV4(),
		Timestamp: time.Now(),
		Headers:   headers,
		Body: string(body),
	}
```